bool joints
bool dest
---
float32[] data